src/wh_server_keystore.c

```diff
@@ -89,6 +117,52 @@ int hsmCacheKey(whServerContext* server, whNvmMetadata* meta, uint8_t* in)
     return 0;
 }
 
+/* try to put the specified key into cache if it isn't already, return index */
+int hsmFreshenKey(whServerContext* server, whKeyId keyId)
+{
+    int ret = 0;
+    int i;
+    int foundIndex = -1;
+    uint32_t outSz = WOLFHSM_KEYCACHE_BUFSIZE;
+    whNvmMetadata meta[1] = {0};
+    if (server == NULL || keyId == WOLFHSM_ID_ERASED)
+        return WH_ERROR_BADARGS;
+    for (i = 0; i < WOLFHSM_NUM_RAMKEYS; i++) {
+        /* check for empty slot or rewrite slot */
+        if ((foundIndex == -1 &&
+            server->cache[i].meta->id == WOLFHSM_ID_ERASED) ||
+            server->cache[i].meta->id == keyId) {
+            foundIndex = i;
+            if (server->cache[i].meta->id == keyId)
+                return i;
```

Comment:
no, that `if` case happens when the key is found to be already in the cache. The `if` statement this one is nested in checks for keyId match or empty, and will continue checking even if it finds an empty spot just in case there's an empty spot before the key.